src/modules/launcher/Microsoft.Launcher/dllmain.cpp

```diff
@@ -75,12 +78,18 @@ class Microsoft_Launcher : public PowertoyModuleIface
     // Handle to event used to invoke the Runner
     HANDLE m_hEvent;
 
+    std::shared_ptr<Logger> logger;
+
 public:
     // Constructor
     Microsoft_Launcher()
     {
         app_name = GET_RESOURCE_STRING(IDS_LAUNCHER_NAME);
         app_key = LauncherConstants::ModuleKey;
+        std::filesystem::path logFilePath(PTSettingsHelper::get_module_save_folder_location(this->app_key));
```

Comment:
I think we should wrap this stuff into try/catch. Ideally, we'd use `null_logger` in the `catch` block, so we don't crash later when attempting to log.

Reply:
That part should not fail. Could you please provide cases when it fails and we still want to run our application?

Reply:
I was thinking about the case where the file couldn't be created for some reason

Reply:
I think If a file can not be created in a dedicated AppData folder by the runner process then we should fail the runner or fail corresponding module loading. Does it make sense?

Reply:
If sink or logger can not be created we create stdout logger and write error to stderr stream.